gwpy/cli/cliproduct.py

```diff
@@ -724,6 +731,49 @@ def run(self):
                 show_error = False
             self.plot_num += 1
 
+    def add_segs(self, args):
+        """ If requested add DQ segments"""
+        std_segments = [
+                '{ifo}:DMT-GRD_ISC_LOCK_NOMINAL:1',
+                '{ifo}:DMT-DC_READOUT_LOCKED:1',
+                '{ifo}:DMT-CALIBRATED:1',
+                '{ifo}:DMT-ANALYSIS_READY:1'
+            ]
+        segments = list()
+        if hasattr(args, 'std_seg'):
+            if args.std_seg:
+                segments = std_segments
+            if args.seg:
+                for seg in args.seg:
+                    # NB: args.seg may be list of lists
+                    segments += seg
+
+            chan = args.chan[0][0]
+            m = re.match('^([A-Za-z][0-9]):', chan)
+            ifo = m.group(1) if m else '?:'
+
+            start = None
+            end = 0
+            for ts in self.timeseries:
+                if start:
+                    start = min(ts.t0, start)
+                    end = max(ts.t0+ts.duration, end)
+                else:
+                    start = ts.t0
+                    end = start + ts.duration
+
+            for segment in segments:
+                seg_name = segment.replace('{ifo}', ifo)
+                try:
+                    seg_data = DataQualityFlag. \
+                        query_dqsegdb(seg_name, start, end,
+                                      url='https://segments.ligo.org')
```

Comment:
Can you please restructure this as

```python
                    seg_data = DataQualityFlag.query_dqsegdb(
                        seg_name,
                        start,
                        end,
                    )
```
and do not include a hardcoded `url` argument, the method itself has logic to auto-select the default server which allows the user to optimise their selection across the board.

Reply:
that does solve the bare except problem